x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1327,7 +1371,8 @@ void fillCacheRange(
             @Nullable SourceInputStreamFactory streamFactory,
             int relativePos,
             int length,
-            IntConsumer progressUpdater
+            IntConsumer progressUpdater,
+            ActionListener<Void> completion
```

Comment:
can we add a javadoc about the completion listener? also, I think that this won't throw an `IOException` anymore?

Reply:
++ rename to `completionListener` everywhere